src/app/api/widget-execution/route.tsx

```diff
@@ -33,71 +33,102 @@ import { getSiteUrl } from 'src/const/urls';
 import { fetchChainData } from 'src/utils/image-generation/fetchChainData';
 import { fetchTokenData } from 'src/utils/image-generation/fetchTokenData';
 import { parseSearchParams } from 'src/utils/image-generation/parseSearchParams';
+import {
+  widgetExecutionSchema,
+  type WidgetExecutionParams,
+} from 'src/utils/image-generation/widgetSchemas';
 
 const WIDGET_IMAGE_WIDTH = 416;
 const WIDGET_IMAGE_HEIGHT = 432;
 const WIDGET_IMAGE_SCALING_FACTOR = 2;
 
 export async function GET(request: Request) {
-  const {
-    fromChainId,
-    toChainId,
-    fromToken,
-    toToken,
-    isSwap,
-    theme,
-    amount,
-    highlighted,
-  } = parseSearchParams(request.url);
+  try {
+    const params = parseSearchParams<WidgetExecutionParams>(
```

Comment:
Should be handle in the same way as the above file to keep consistency https://github.com/jumperexchange/jumper-exchange/pull/1841/files#diff-a2779d950e5a48246e8ecacd930c3d762f5302e7cd0cb7a79d1fbdfcfb2f94c8R41